### DISCUSSION TITLE
Setup in sketch for different TFT's?

### DISCUSSION THREAD

**Discussion by PaulVdBo** [Bodmer/TFT_eSPI#2966] [Q&A - General]
Hey Bodmer !
First of all : congrats for your unbelievable library, and even more for your never ending support ! Thanks, thanks thanks !!!
Now my question : I program for different TFT's. Is there a possibility to "incorporate" a specific TFT in the main .ino instead of changing the user_setup  file every time ?
thanks in advance !!!

**Comment by Bodmer** [comment]
No, this is a choice I made a long time ago to keep my life simple! All examples will run on my different hardware setups after a very quick and simple edit of the User_Setup_Select.h file (insert // and remove //). Then I can run any of the many test sketches and examples without modifying them. A desktop shortcut means loading the header to edit is trivial.  You will probably find libraries that provide configuration in the sketches provide very few useful examples due to the hassle of changing every test sketch for different hardware setups. If you use different processors (or boards which have a defined identifier) then ifdef... can be used in the header so different processors/boards use different setup files.

Using PlatformIO allows projects to have their own configuration settings in the platformio.ini file, so that may be an alternative approach.
